### PR TITLE
[WIP] Support NHWC layout in ChainerX

### PR DESCRIPTION
**Pull request by take-cheeze**
ref: https://github.com/chainer/chainer/pull/7375 , https://github.com/chainer/chainer/pull/7620

TODOs
- [ ] Make layout specifying easier
- [ ] Pool support
- [ ] non-4D tensor

### CODE REVIEW

**Comment by niboshi** [comment]
@take-cheeze I'm sorry for unattendance.
Now that #7620 has been merged, I think it's ready to add NHWC support to ChainerX.
Are you willing to continue this PR? We can take it over if you don't have time.

**Comment by take-cheeze** [comment]
@niboshi 
Thank you for reminding me.
Unfortunately I've forgot most of thing when I started this I'll be happy if you can take over this!

**Comment by emcastillo** [comment]
I can try to take over this in two steps.

1st, allow convolution and batchnorm of chainerx to use nhwc layout. 
This will allow using it directly from chainer.

2nd, add layout support to the chainerx arrays as in the chainer NHWC support.

**Comment by niboshi** [comment]
We should be careful about the interface.

I don't think generic interface like `chainerx.batch_norm` should have explicit `layout` argument, because it's a CUDA-specific matter.

**Comment by emcastillo** [comment]
It is a very good point, I assume that since cupy just maps cuda & cudnn to python, it is ok to expose it in that case but not with chainerx ...

**Comment by niboshi** [comment]
We're no longer merging feature PRs.
Let me close the PR.
